### PR TITLE
mgr/cephadm: check existence of the RADOS pool

### PR DESCRIPTION
**Pull request by mgfritch**
Check existence of the RADOS pool during `orch reconfig/redeploy`:

```
$ bin/ceph orch reconfig nfs.foo
Error EINVAL: Traceback (most recent call last):
  File "/usr/share/ceph/mgr/mgr_module.py", line 1171, in _handle_command
    return self.handle_command(inbuf, cmd)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 138, in handle_command
    return dispatch[cmd['prefix']].call(self, cmd, inbuf)
  File "/usr/share/ceph/mgr/mgr_module.py", line 311, in call
    return self.func(mgr, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 100, in <lambda>
    wrapper_copy = lambda *l_args, **l_kwargs: wrapper(*l_args, **l_kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 89, in wrapper
    return func(*args, **kwargs)
  File "/usr/share/ceph/mgr/orchestrator/module.py", line 1088, in _service_action
    raise_if_exception(completion)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 658, in raise_if_exception
    raise e
rados.ObjectNotFound: [errno 2] RADOS object not found (RADOS object not found (error opening pool 'b'cephfs.a.dat''))
```

Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/services/nfs.py [hunk 1]
```diff
@@ -111,10 +114,9 @@ def get_set_cmd_dicts(out: str) -> List[dict]:
 
 class NFSGanesha(object):
     def __init__(self,
-                 mgr,
-                 daemon_id,
-                 spec):
-        # type: (cephadm.CephadmOrchestrator, str, NFSServiceSpec) -> None
+                 mgr: "CephadmOrchestrator",
+                 daemon_id: str,
+                 spec: NFSServiceSpec) -> None:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/services/nfs.py:119]: fyi, there is a tool called `com2ann` which worked pretty well for me.

File: src/pybind/mgr/cephadm/services/nfs.py [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
 
-import orchestrator
-from orchestrator import OrchestratorError
-from orchestrator import DaemonDescription
-
-import cephadm
-from .. import utils
-
-from .cephadmservice import CephadmService, CephadmDaemonSpec
+import rados
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/services/nfs.py:9]:
```python
import logging
from typing import TYPE_CHECKING, Dict, Optional, Tuple, Any, List, Set, cast

from ceph.deployment.service_spec import NFSServiceSpec
import rados

from orchestrator import OrchestratorError, DaemonDescription

from cephadm import utils
from cephadm.services.cephadmservice import CephadmService, CephadmDaemonSpec
```

I'd import

1. standard libs
2. 3rd party libs
3. our own python libs
4. other mgr modules
5. our own mgr module

File: src/pybind/mgr/cephadm/services/nfs.py (hunk 1 above)
Comment (mgfritch) [reply src/pybind/mgr/cephadm/services/nfs.py:120]: > fyi, there is a tool called `com2ann` which worked pretty well for me.

neat! that does make these much easier to resolve.

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-08-07_14:25:21-rados:cephadm-wip-swagner-testing-2020-08-07-1340-distro-basic-smithi/

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-08-08_13:50:13-rados-wip-kefu-testing-2020-08-08-1822-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-09_03:30:55-rados-wip-kefu-testing-2020-08-08-1822-distro-basic-smithi/

failure in upgrade test also exists in master

- https://pulpito.ceph.com/kchai-2020-08-09_05:31:11-rados-master-distro-basic-smithi/

**Pull request merged**